lib/src/widgets/nes_progress_bar.dart

```diff
@@ -0,0 +1,201 @@
+import 'package:flutter/material.dart';
+import 'package:nes_ui/nes_ui.dart';
+import 'package:nes_ui/src/extensions/color.dart';
+
+/// Enum with the possible styles of [NesProgressBar].
+enum NesProgressBarStyle {
+  /// The progress will fill 1 pixel at a time. (Default value)
+  pixel,
+
+  /// The progress will fill smoothly.
+  smooth,
+}
+
+/// {@template nes_progress_bar}
+/// A widget that displays a progress amount.
+/// {@endtemplate}
+class NesProgressBar extends StatefulWidget {
+  /// Creates a progress bar.
+  ///
+  /// {@macro nes_ui.NesProgressBar}
+  const NesProgressBar({
+    required this.value,
+    this.style = NesProgressBarStyle.pixel,
+    this.width,
+    this.label,
+    super.key,
+  });
+
+  /// The value for this progress bar.
+  ///
+  /// A value of 0.0 means no progress and 1.0 means that progress is complete.
+  /// The value will be clamped to be in the range 0.0-1.0.
+  ///
+  /// This value will be read out by screen readers expressed as a percentage.
+  /// i.e. `0.1` will become '10%'.
+  final double value;
+
+  /// The style of fill to use for the progress bar.
+  ///
+  /// Defaults to [NesProgressBarStyle.pixel]
+  final NesProgressBarStyle style;
+
+  /// The width of this progress bar.
+  ///
+  /// If null, the progress bar will fill the parent widget.
+  final double? width;
+
+  /// The label for this progress bar.
+  ///
+  /// This value indicates the purpose of the progress bar, and will be
+  /// read out by screen readers to indicate the purpose of this progress bar.
+  final String? label;
+
+  @override
+  State<NesProgressBar> createState() => _NesProgressBarState();
+}
+
+class _NesProgressBarState extends State<NesProgressBar> {
+  @override
+  void didUpdateWidget(NesProgressBar oldWidget) {
+    super.didUpdateWidget(oldWidget);
+  }
```

Comment:
```suggestion
```

Reply:
Also, does this needs to be a stateful widget at all? It doesn't seems to have any state being managed, just a build method that will render based on the properties from the widget.

Reply:
removed stateful widget.